### PR TITLE
Updated gmp/gvm_connection for modification of tasks/targets

### PR DESCRIPTION
**Pull request by jschwartz27**
Updated gmp.py and gvm_connection.py scripts allowing modification of tasks and targets

### CODE REVIEW

File: gmp/gmp.py [hunk 1]
```diff
@@ -1269,10 +1271,125 @@ def modifyTagCommand(self, tag_id, kwargs):
...
+        
+        reverse_lookup_unify = kwargs.get('reverse_lookup_unify', '')
+        if reverse_lookup_unify:
+            reverse_lookup_unify = '<reverse_lookup_unify>%s</reverse_lookup_unify>' % reverse_lookup_unify
+
+        port_range = kwargs.get('port_range', '')
+        if port_range:
+            port_range = '<port_range>%s</port_range>' % port_range
+
+        if 'port_list' in kwargs:
```
Comment (wiegandm) [inline gmp/gmp.py]: Note that this will lead to `port_list` being undefined below if there is no `port_list` key in the kwargs.

File: gmp/gmp.py [hunk 2]
```diff
@@ -1269,10 +1271,125 @@ def modifyTagCommand(self, tag_id, kwargs):
...
+            port_list = kwargs.get('port_list')
+            if 'id' in port_list:
+                port_list = '<port_list id="%s"/>' % port_list
+
+        return '<modify_target target_id="{0}">{1}{2}{3}{4}{5}{6}  \
```
Comment (wiegandm) [inline gmp/gmp.py]: The `format` statement contains 12 placeholders in the string, but there are only 11 parameters passed to the method.

There is either one placeholder too many or a missing parameter.

File: gmp/gmp.py [hunk 3]
```diff
@@ -1269,10 +1271,136 @@ def modifyTagCommand(self, tag_id, kwargs):
...
+
+        alert = kwargs.get('alert', '')
+        if alert:
+            alert = '<alert id="%s"/>' % (alert)
+
+        observers = kwargs.get('observers', '')
+        if observers:
+            observers = '<observers>%s</observers>' % observers
+
+        preferences = kwargs.get('preferences', '')
```
Comment (rgrewe) [inline gmp/gmp.py]: Could you give an example how the values for preferences are given to this function?

File: gmp/gmp.py [hunk 4]
```diff
@@ -1269,10 +1271,136 @@ def modifyTagCommand(self, tag_id, kwargs):
...
+        file = kwargs.get('file', '')
+        if file:
+            file_name = file['name']
+            file_action = file['action']
+            if file_action != "update" and file_action !="remove" :
+                raise ValueError('action can only be "update" or "remove"!')
+
+            file = '<file name="%s" action="%s"/>' % (file_name, file_action)
+
+        print('<modify_task task_id="{0}">' \
```
Comment (wiegandm) [inline gmp/gmp.py]: I guess the `print` is a debug statement which ought to be removed.

File: gmp/gmp.py (hunk 3 above)
Comment (jschwartz27) [reply gmp/gmp.py]: As a dictionary where, preferences = {"scanner_name": ["name1", "name2"], "value": ["value1", "value2"]}.
perhaps there is a more efficient or desirable way of doing this?

**Pull request merged**